R/sample.R

```diff
@@ -156,16 +213,62 @@ combine_chains <- function(res) {
                 "runner")))
   }
 
-  ret <- list(pars = pars,
-              density = density,
-              details = details,
-              chain = chain)
-  class(ret) <- "mcstate_samples"
-  ret
+  samples <- list(pars = pars,
+                  density = density,
+                  details = details,
+                  chain = chain)
+  class(samples) <- "mcstate_samples"
+  samples
+}
+
+
+## This is absolutely terrible, but it will get there.
```

Comment:
Yeah - the next few lines I am somewhat treating as a blackbox, but the tests are very clear and working!